File: Paint_FrontEnd/src/app/Shapes/ishape.ts
import { __asyncDelegator } from "tslib";

export abstract class IShape {
    private _x: number;

    private _y: number;

    private _stroke: string;

    private _strokeWidth: number;
    
    private _fill: string;
    
    private _rotate: number;

    private _draggable: boolean;
    
    private _id: string;
    
    private _type: string = "";

    private _update: boolean;

    private _deleteflag: boolean;

    constructor(x: number, y: number, stroke: string, strokeWidth: number, fill: string, rotate: number, draggable: boolean, id: string, update: boolean, deleteflag: boolean){
        this._x = x;
        this._y = y;
        this._stroke = stroke;
        this._strokeWidth = strokeWidth;
        this._fill = fill;
        this._rotate = rotate;
        this._draggable = draggable;
        this._id = id;
        this._update = update;
        this._deleteflag = deleteflag;
    }

    public get x(): number {
        return this._x;
    }
    public set x(value: number) {
        this._x = value;
    }
    public get y(): number {
        return this._y;
    }
    public set y(value: number) {
        this._y = value;
    }
    public get stroke(): string {
        return this._stroke;
    }
    public set stroke(value: string) {
        this._stroke = value;
    }
    public get strokeWidth(): number {
        return this._strokeWidth;
    }
    public set strokeWidth(value: number) {
        this._strokeWidth = value;
    }
    public get fill(): string {
        return this._fill;
    }
    public set fill(value: string) {
        this._fill = value;
    }
    public get rotate(): number {
        return this._rotate;
    }
    public set rotate(value: number) {
        this._rotate = value;
    }
    public get draggable(): boolean {
        return this._draggable;
    }
    public set draggable(value: boolean) {
        this._draggable = value;
    }
    public get id(): string {
        return this._id;
    }
    public set id(value: string) {
        this._id = value;
    }
    public get update(): boolean {
        return this._update;
    }
    public set update(value: boolean) {
        this._update = value;
    }
    public get type(): string {
        return this._type;
    }
    public set type(value: string) {
        this._type = value;
    }
    public get deleteflag(): boolean {
        return this._deleteflag;
    }
    public set deleteflag(value: boolean) {
        this._deleteflag = value;
    }
    
}

File: Paint_FrontEnd/src/main/java/com/paint/paint_backend/Shapes/IShape.java
package com.paint.paint_backend.Shapes;

public class IShape {
    double _x, _y, _rotate;
    String _stroke;
    double _strokeWidth;
    String _fill;
    String _id;

    String _type;
    boolean _draggable, _update, _deleteflag;
    IShape(double x, double y, String stroke, double strokeWidth, String fill, double rotate, boolean draggable, String id, boolean update, boolean deleteflag){
        this._x = x;
        this._y = y;
        this._stroke = stroke;
        this._strokeWidth = strokeWidth;
        this._fill = fill;
        this._rotate = rotate;
        this._draggable = draggable;
        this._id = id;
        this._update = update;
        this._deleteflag = deleteflag;
    }
    public double get_x() {
        return _x;
    }

    public void set_x(double _x) {
        this._x = _x;
    }

    public double get_y() {
        return _y;
    }

    public void set_y(double _y) {
        this._y = _y;
    }

    public double get_rotate() {
        return _rotate;
    }

    public void set_rotate(double _rotate) {
        this._rotate = _rotate;
    }

    public String get_stroke() {
        return _stroke;
    }

    public void set_stroke(String _stroke) {
        this._stroke = _stroke;
    }

    public String get_fill() {
        return _fill;
    }

    public void set_fill(String _fill) {
        this._fill = _fill;
    }

    public String get_id() {
        return _id;
    }

    public void set_id(String _id) {
        this._id = _id;
    }

    public boolean is_draggable() {
        return _draggable;
    }

    public void set_draggable(boolean _draggable) {
        this._draggable = _draggable;
    }

    public boolean is_update() {
        return _update;
    }

    public void set_update(boolean _update) {
        this._update = _update;
    }

    public boolean is_deleteflag() {
        return _deleteflag;
    }

    public void set_deleteflag(boolean _deleteflag) {
        this._deleteflag = _deleteflag;
    }

    public double get_strokeWidth() {
        return _strokeWidth;
    }

    public void set_strokeWidth(double _strokeWidth) {
        this._strokeWidth = _strokeWidth;
    }

    public String get_type() {
        return _type;
    }

    public void set_type(String _type) {
        this._type = _type;
    }
}
